tools/beman-tidy/beman_tidy/lib/checks/beman_standard/repository.py

```diff
@@ -71,4 +74,42 @@ def fix(self):
         )
 
 
-# TODO REPOSITORY.DISALLOW_GIT_SUBMODULES
+@register_beman_standard_check("REPOSITORY.DISALLOW_GIT_SUBMODULES")
+class RepositoryDisallowGitSubmodulesCheck(FileBaseCheck):
+    def __init__(self, repo_info, beman_standard_check_config):
+        super().__init__(repo_info, beman_standard_check_config, ".gitmodules")
+
+    def pre_check(self):
+        # Need to override this, because REPOSITORY.DISALLOW_GIT_SUBMODULES is conditional
+        return True
+
+    def check(self):
+        if self.path.exists():
+            content = self.read()
+
+            # Regex pattern to match "wg21" submodule
+            regex = re.compile(
+                textwrap.dedent(r"""
+                    ^\[submodule \"(?:.+?/)?wg21\"]
+                    \tpath = (?:.+?/)?wg21
```

Comment:
```suggestion
                    \tpath = (?:.+?/)?
```
It does not need to be wg21 -> https://github.com/bemanproject/optional/blob/main/.gitmodules#L2. It's a local path.